src/main/java/com/yscope/clp/compressorfrontend/EncodedMessage.java

```diff
@@ -51,11 +57,38 @@ public String[] getDictionaryVarsAsStrings() {
     return dictVars;
   }
 
+  public void computeFlattenedDictionaryVars() {
```

Comment:
Instead of adding this method to `EncodeMessage`'s API, how about we extract the dictionary flattening logic from `MessageDecoder` into a utility class and use that utility class both in `MessageDecoder` and the unit tests?